spec/lib/msf/core/post/vcenter/database_spec.rb

```diff
@@ -0,0 +1,209 @@
+vpxuser|*ZdvmNiLEXzZL/uhdW6Zb4Px4RR72iD+xftdA0n9hJ8xpFNJW/axpyKMQ8BJWIFTzzoxQnAm2PaX486yExLX7qg==|192.168.20.20|192.168.20.15|test1.local')
+      expect(subject.query_vpx_creds('test', 'test', 'test')).to eq([
+        {
+          'dns_name' => '192.168.20.10',
+          'encrypted_password' =>
+            '*tktZGW50GH4BEOXyWCr9WTu2PSMGWSvcqEsuAMnwcNuFO/rQPRsOyygRRY/WaM3IOI/BrqcThiaiM3j4Jw+KtA==',
+          'ip_address' => '192.168.20.10',
+          'local_ip' => '192.168.20.20',
+          'user' => 'vpxuser'
+        },
+        {
+          'dns_name' => 'test1.local',
+          'encrypted_password' =>
+          '*ZdvmNiLEXzZL/uhdW6Zb4Px4RR72iD+xftdA0n9hJ8xpFNJW/axpyKMQ8BJWIFTzzoxQnAm2PaX486yExLX7qg==',
+          'ip_address' => '192.168.20.15',
+          'local_ip' => '192.168.20.20',
+          'user' => 'vpxuser'
+        }
+      ])
+    end
+    it 'with a valid entry and symkey' do
+      # combination of https://github.com/rapid7/metasploit-framework/pull/16465#issuecomment-1117587575
+      # and https://github.com/shmilylty/vhost_password_decrypt
+      allow(subject).to receive(:command_exists?).and_return(true)
+      allow(subject).to receive(:cmd_exec).and_return('vpxuser|*SN2otuvNvGRSC29lxhU4XQbgNOMyVawGF4UHA38w2zq59tX0WzkgkQTNBJSJpHvBvkYwyiR8xNAv1oquEOOLvQ==|192.168.20.20|192.168.20.15|test1.local')
+      expect(subject.query_vpx_creds('test', 'test', 'test', 'f1d0d054e43ac880809c354cec681b3433e36fc4ea6b1480de05b7b86c3506cd')).to eq([
+        {
+          'dns_name' => 'test1.local',
+          'encrypted_password' =>
+          '*SN2otuvNvGRSC29lxhU4XQbgNOMyVawGF4UHA38w2zq59tX0WzkgkQTNBJSJpHvBvkYwyiR8xNAv1oquEOOLvQ==',
+          'ip_address' => '192.168.20.15',
+          'local_ip' => '192.168.20.20',
+          'user' => 'vpxuser',
+          'decrypted_password' => '-KOU.80J\I0n\Pcqya3F0af=z5Ix-5.u'
+        }
+      ])
+    end
+    it 'with a valid entry and bad symkey' do
+      # combination of https://github.com/rapid7/metasploit-framework/pull/16465#issuecomment-1117587575
+      # and https://github.com/shmilylty/vhost_password_decrypt
+      allow(subject).to receive(:command_exists?).and_return(true)
+      allow(subject).to receive(:cmd_exec).and_return('vpxuser|*SN2otuvNvGRSC29lxhU4XQbgNOMyVawGF4UHA38w2zq59tX0WzkgkQTNBJSJpHvBvkYwyiR8xNAv1oquEOOLvQ==|192.168.20.20|192.168.20.15|test1.local')
+      expect(subject.query_vpx_creds('test', 'test', 'test', 'bad0d054e43ac880809c354cec681b3433e36fc4ea6b1480de05b7b86c3506cd')).to eq([
+        {
+          'dns_name' => 'test1.local',
+          'encrypted_password' =>
+          '*SN2otuvNvGRSC29lxhU4XQbgNOMyVawGF4UHA38w2zq59tX0WzkgkQTNBJSJpHvBvkYwyiR8xNAv1oquEOOLvQ==',
+          'ip_address' => '192.168.20.15',
+          'local_ip' => '192.168.20.20',
+          'user' => 'vpxuser'
+        }
+      ])
+    end
+  end
+
+  # XXX need to add a real user test
```

Comment:
they were in the last one, and transferred into this one. I can't test some of the stuff from the original module since I don't have the env, so just marking ones that need to be further tested by someone who has a more robust setup than myself